### PR TITLE
feat(storage): use variable size encoding to compress rle count

### PR DESCRIPTION
**Pull request by xinchengforgit**
Signed-off-by: xinchengforgit 2419743144@qq.com

This PR try to solve #531.
I have some basic thoughts about variable size encoding, but there are some tests that I don't know how to modify, please give me some suggestions.

I use variable size encoding, and replace the u16 with u32, so that we can support a larger range number while saving the used space.

### CODE REVIEW

File: src/storage/secondary/block/rle_block_builder.rs [hunk 1]
```diff
@@ -2,34 +2,109 @@
...
 
-use bytes::BufMut;
+use bytes::{Buf, BufMut};
+use prost::DecodeError;
 use risinglight_proto::rowset::BlockStatistics;
 
 use super::BlockBuilder;
 use crate::array::Array;
 
+pub fn encode_32<B>(mut value: u32, buf: &mut B)
```
Comment (skyzh) [inline src/storage/secondary/block/rle_block_builder.rs:12]: Would you please write docstring for this function? Also, would you please explain a little bit about the implementation? (e.g. borrowed from other products? Or some general ideas about this)

File: src/storage/secondary/block/rle_block_builder.rs [hunk 2]
```diff
@@ -166,7 +246,7 @@ mod tests {
             .iter()
             .cycle()
             .cloned()
-            .take(u16::MAX as usize * 2)
+            .take(u32::MAX as usize + 1)
```
Comment (skyzh) [inline src/storage/secondary/block/rle_block_builder.rs:249]: Will this cause test to be running very slow? If so, we can avoid testing it.

File: src/storage/secondary/block/rle_block_builder.rs [hunk 3]
```diff
@@ -88,9 +163,14 @@ where
         }
         self.rle_counts.push(self.cur_count);
         encoded_data.put_u32_le(self.rle_counts.len() as u32);
+        let offset = encoded_data.len();
+        encoded_data.put_u32_le(0xffffffff); // placeholder
```
Comment (skyzh) [inline src/storage/secondary/block/rle_block_builder.rs:167]: Do we really need this length? We can read until the end when decoding given the number of RLE count item.

**Review by skyzh — COMMENTED** [review]
Rest LGTM, good work! Please help fix fmt. See developer docs for more information.

File: src/storage/secondary/block/rle_block_builder.rs (hunk 1 above)
Comment (xinchengforgit) [reply src/storage/secondary/block/rle_block_builder.rs:13]: Yeah, In fact, I borrow the idea from the prost::encode_varint.
This function encodes u32 to multiple u8 , and for every u8, the highest bit tells whether this u8 belongs to the same u32.

File: src/storage/secondary/block/rle_block_builder.rs (hunk 2 above)
Comment (xinchengforgit) [reply src/storage/secondary/block/rle_block_builder.rs:251]: I think it does cause test to be running very slow. But I don't know how to change the test myself.

File: src/storage/secondary/block/rle_block_builder.rs (hunk 1 above)
Comment (skyzh) [reply src/storage/secondary/block/rle_block_builder.rs:13]: Please add license header of prost in this file.

File: src/storage/secondary/block/rle_block_builder.rs [hunk 4]
```diff
@@ -20,9 +94,11 @@ where
     A::Item: PartialEq,
 {
     block_builder: B,
-    rle_counts: Vec<u16>,
+    rle_counts: Vec<u32>,
```
Comment (skyzh) [inline src/storage/secondary/block/rle_block_builder.rs:97]: Please update comments on L88

**Review by skyzh — APPROVED** [review]
Rest LGTM, good work!

**Pull request merged**